### PR TITLE
Allow declaring dependent services for a JobHandler

### PR DESCRIPTION
**Pull request by frojasg**
We may need dependant service before we start consuming events in a JobHandler.

### CODE REVIEW

File: misk-aws/src/main/kotlin/misk/jobqueue/sqs/AwsSqsJobHandlerModule.kt [hunk 1]
```diff
@@ -14,17 +17,21 @@ import kotlin.reflect.KClass
...
-  private val handler: KClass<T>
+  private val handler: KClass<T>,
+  private val dependsOn: List<Key<out Service>> = listOf()
 ) : KAbstractModule() {
   override fun configure() {
     newMapBinder<QueueName, JobHandler>().addBinding(queueName).to(handler.java)
-    install(ServiceModule<AwsSqsJobHandlerSubscriptionService>()
-        .dependsOn<SqsJobConsumer>())
+    install(ServiceModule(AwsSqsJobHandlerSubscriptionService::class.toKey(),
+        dependsOn + listOf(SqsJobConsumer::class.toKey())))
```
Comment (swankjesse) [inline misk-aws/src/main/kotlin/misk/jobqueue/sqs/AwsSqsJobHandlerModule.kt]: FYI @keeferrourke

Comment (swankjesse) [reply misk-aws/src/main/kotlin/misk/jobqueue/sqs/AwsSqsJobHandlerModule.kt]: I’d love to give you a mechanism to do this elsewhere

**Comment by swankjesse** [comment]
... I’m trying to think of what the obvious test looks like, and I think it probably asserts something about the ServiceGraph. @keeferrourke is that easy?

My recommendation is you merge as is, and I borrow some of @keeferrourke’s time to add the test. Cause testing this might not be as easy as it should be.

**Comment by keeferrourke** [comment]
> ... I’m trying to think of what the obvious test looks like, and I think it probably asserts something about the ServiceGraph. @keeferrourke is that easy?
> 
> My recommendation is you merge as is, and I borrow some of @keeferrourke’s time to add the test. Cause testing this might not be as easy as it should be.

That sounds good to me!

**Comment by ryanhall07** [comment]
> ... I’m trying to think of what the obvious test looks like, and I think it probably asserts something about the ServiceGraph. 

My initial thought was a test specific to the jobqueue. You create a fake dependent service that flips some boolean and in the job handler you verify that boolean has been flipped.

File: misk-aws/src/main/kotlin/misk/jobqueue/sqs/AwsSqsJobHandlerModule.kt (hunk 1 above)
Comment (keeferrourke) [reply misk-aws/src/main/kotlin/misk/jobqueue/sqs/AwsSqsJobHandlerModule.kt]: #1055

**Comment by swankjesse** [comment]
I think this can be closed. We fixed it so that all services are running before any jobs run, which makes the contract between jobs and their handlers very straightforward.

**Comment by frojasg** [comment]
closing in favor of #1055

**Pull request closed without merging**